.github/workflows/reusable_mix_project.yml

```
@@ -29,11 +27,26 @@ jobs:
         with:
           ref: ${{ github.head_ref }}
 
-      - name: Set up Elixir ${{ env.ELIXIR_VERSION }} and OTP ${{ env.OTP_VERSION }}
-        uses: erlef/setup-elixir@v1
+      - name: Setup asdf
+        uses: asdf-vm/actions/setup@v1
+
+      - name: Cache asdf
+        uses: actions/cache@v2
         with:
-          otp-version: ${{ env.OTP_VERSION }}
-          elixir-version: ${{ env.ELIXIR_VERSION }}
+          path: |
+            /home/runner/.asdf
```

It's the same suggestion as Hoang. It can be on the same line

```suggestion
          path: /home/runner/.asdf
```